packages/flutter/lib/src/widgets/raster_widget.dart

```diff
@@ -0,0 +1,401 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'media_query.dart';
+
+/// A controller for the [RasterWidget] that controls when the child image is displayed
+/// and when to regenerated the child image.
+///
+/// When the value of [rasterize] is true, The [RasterWidget] will convert the child scene into
+/// an image. This image will be drawn in place of the children until [rasterize] is false.
+/// Any changes to the widget tree that would normally cause it to repaint are ignored.
+///
+/// To force [RasterWidget] to recreate the child image, call [clear].
+class RasterWidgetController extends ChangeNotifier {
+  /// Create a new [RasterWidgetController].
+  ///
+  /// By default, [rasterize] is `false` and cannot be `null`.
+  RasterWidgetController({
+    bool rasterize = false
+  }) : _rasterize = rasterize;
+
+  /// Reset the raster held by any listening [RasterWidget].
+  ///
+  /// This has no effect if [rasterize] is `false`.
+  void clear() {
+    notifyListeners();
+  }
+
+  /// Whether a rasterized version of this render objects child is drawn in
+  /// place of the child.
+  bool get rasterize => _rasterize;
+  bool _rasterize;
+  set rasterize(bool value) {
+    if (value == rasterize) {
+      return;
+    }
+    _rasterize = value;
+    notifyListeners();
+  }
+}
+
+/// A delegate which the [RasterWidget] can use to fallback to regular rendering
+/// if a platform view is present in the layer tree.
+///
+/// Consumers of [RasterWidget] should almost never use this delegate. For the most part,
+/// the raster widget only functions as a performance improvement. If a platform view is
+/// present, the performance improving qualities aren't possible and using this API is
+/// pointless.
+///
+/// Instead, this interface is useful if a generic/reusable widget is being created which
+/// may include a platform view and it needs to handle this transparently. For example, the
+/// framework uses this for the zoom page transition so that navigating to a page shows the same
+/// animation whether or not there is a platform view.
+abstract class RasterWidgetFallbackDelegate {
```

Comment:
nitnit: move this closer to the regular delegate?

Reply:
Done